### PR TITLE
add support for BerkeleyGW 4.0

### PR DESCRIPTION
**Pull request by migueldiascosta**

### CODE REVIEW

**Comment by migueldiascosta** [comment]
@spackbot fix style

**Comment by migueldiascosta** [comment]
@spackbot run pipeline

File: var/spack/repos/builtin/packages/berkeleygw/package.py [hunk 1]
```diff
@@ -147,19 +153,34 @@ def edit(self, spec, prefix):
...
+        for test in si_epm_subspace_tests:
             filter_file(
                 "Precision : 6e-15",
                 "Precision : 7e-15",
                 join_path("testsuite", "Si-EPM_subspace", test + ".test"),
             )
         filter_file("Precision : 8e-15", "Precision : 9e-15", "testsuite/GaAs-EPM/GaAs.test")
 
+        if self.version < Version("3.1.0"):
+            # np.int alias was removed from numpy
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/berkeleygw/package.py:179]: (question) based on this comment, should this be based on the version of `py-numpy` that is used as well?

**Review by scheibelp — COMMENTED** [review]
I have a question about the NumPy alias patch

Patching to the tolerance of the tests is interesting. I'm curious if this is system/compiler dependent, would you be able to mention those in this PR?

File: var/spack/repos/builtin/packages/berkeleygw/package.py (hunk 1 above)
Comment (migueldiascosta) [reply var/spack/repos/builtin/packages/berkeleygw/package.py:179]: it is true that the `numpy.int` alias would still work for `numpy` < 1.24, but the `int` builtin always works, so this seems simpler?

(https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations, https://numpy.org/devdocs/release/1.24.0-notes.html#expired-deprecations)

Comment (scheibelp) [reply var/spack/repos/builtin/packages/berkeleygw/package.py:179]: That sounds good - thanks for the response!

**Review by scheibelp — APPROVED** [review]

**Comment by migueldiascosta** [comment]
> Patching to the tolerance of the tests is interesting. I'm curious if this is system/compiler dependent, would you be able to mention those in this PR?

Yes, I believe it is system/compiler dependent, but one can only know after testing in multiple platforms. If there are similar small differences when running the tests in the CI pipeline I can also patch those (are the tests run in the CI pipeline? these only run when `--test` is passed to `spack install`)

**Comment by scheibelp** [comment]
> (are the tests run in the CI pipeline? these only run when --test is passed to spack install)

They are (the `ci.yaml` config for our pipelines sets this), but note that `berkeleygw` is not in any of our pipelines (do you mean you have one yourself?).

I don't see any `on_attributes(run_tests=True)` methods in here, so I was assuming these tests were run automatically as part of the `berkeleygw` build.

**Comment by migueldiascosta** [comment]
> > (are the tests run in the CI pipeline? these only run when --test is passed to spack install)
> 
> They are (the `ci.yaml` config for our pipelines sets this), but note that `berkeleygw` is not in any of our pipelines (do you mean you have one yourself?).

No, I don't - I thought it would be run as part of the `ci/gitlab-ci` action for this PR
> 
> I don't see any `on_attributes(run_tests=True)` methods in here, so I was assuming these tests were run automatically as part of the `berkeleygw` build.

there is a `make check` target, and afaik spack only invokes that when `--test` is passed (at least that's the behaviour I observe)

**Pull request merged**